src/bandwidthRtc.ts

```diff
@@ -1,132 +1,123 @@
+  /**
+   * Publish media to the Bandwidth WebRTC platform
+   *
+   * This function can publish an existing MediaStream, or it can create and publish a new media stream from MediaStreamConstraints
+   * @param input existing media or specific constraints to publish; optional, defaults to basic audio/video constraints
+   * @param audioLevelChangeHandler handler that can be called when the audio level of the published stream changes (optional)
+   * @param alias stream alias/tag that will be included in subscription events and billing records, should not be PII (optional)
+   */
+  async publish(input?: MediaStreamConstraints | MediaStream, audioLevelChangeHandler?: AudioLevelChangeHandler, alias?: string): Promise<RtcStream> {
+    if (!this.delegate) {
+      throw new BandwidthRtcError("You must call 'connect' before 'publish'");
     }
 
-    let mediaTypes: MediaType[] = [];
-    if (mediaStream.getAudioTracks().length > 0) {
-      mediaTypes.push(MediaType.AUDIO);
-    }
-    if (mediaStream.getVideoTracks().length > 0) {
-      mediaTypes.push(MediaType.VIDEO);
-    }
+    return this.delegate.publish(input, audioLevelChangeHandler, alias);
+  }
 
-    const sdpRequest = await this.signaling.requestToPublish(mediaTypes, alias);
-    const endpointId = sdpRequest.endpointId;
+  /**
+   * Unpublish one or more streams.
+   * @param streams streams to unpublish; leave empty to unpublish all streams
+   */
+  async unpublish(...streams: RtcStream[] | string[]) {
+    if (!this.delegate) {
+      throw new BandwidthRtcError("You must call 'connect' before 'unpublish'");
+    }
 
-    if (audioLevelChangeHandler) {
-      const audioLevelDetector = new AudioLevelDetector({
-        mediaStream: mediaStream,
+    if (this.delegate instanceof BandwidthRtcV2) {
+      streams = (streams as Array<RtcStream | string>).map((stream: RtcStream | string) => {
+        if (typeof stream !== "string") {
+          stream = stream.endpointId;
+        }
+        return stream;
       });
-      audioLevelDetector.on("audioLevelChange", audioLevelChangeHandler);
     }
 
-    const peerConnection = new RTCPeerConnection(RTC_CONFIGURATION);
-    this.setupNewPeerConnection(peerConnection, endpointId, mediaTypes, alias);
-    mediaStream.getTracks().forEach((track) => {
-      var sender = peerConnection.addTrack(track, mediaStream);
-
-      // Inject DTMF into one audio track in the stream
-      if (track.kind === "audio" && !this.localDtmfSenders.has(endpointId)) {
-        this.localDtmfSenders.set(endpointId, new DtmfSender(sender));
-      }
-    });
-
-    this.localPeerConnections.set(endpointId, peerConnection);
-    this.localStreams.set(endpointId, mediaStream);
-
-    await this.negotiateSdp(sdpRequest, peerConnection);
-
-    return {
-      endpointId: endpointId,
-      mediaStream: mediaStream,
-      mediaTypes: mediaTypes,
-      alias: alias,
-    };
-  }
+    console.log("unpublish streams", ...streams);
```

Comment:
Did you intend to still include this? I only ask because it doesn't use the logger

Reply:
Sure didn't, was debugging something. Thanks for catching it.